src/Workspaces/Core/Portable/Options/EditorConfig/IEditorConfigStorageLocationWithKey.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    internal interface IEditorConfigStorageLocationWithKey : IEditorConfigStorageLocation
+    {
+        string KeyName { get; }
+        string GetEditorConfigStringForValue(object value, OptionSet optionSet);
```

Comment:
1. since the sole implementation of IEditorConfigStorageLocation implements this, i don't see the need for multiple itnerfaces here.  This entire interface should just move into IEditorConfigStorageLocation.  

2. naming:  I would just have GetValueString, or GetOptionValueString.

Reply:
> since the sole implementation of IEditorConfigStorageLocation implements this, i don't see the need for multiple itnerfaces here

[NamingStylePreferenceEditorConfigStorageLocation](http://index/?leftProject=Microsoft.CodeAnalysis.Workspaces&leftSymbol=munzsfvtuv45&file=Options%5CEditorConfig%5CNamingStylePreferenceEditorConfigStorageLocation.cs&line=12) also implements `IEditorConfigStorageLocation` but does not implement either of the members added here. I was going to file a separate bug to figure out how to ensure the naming rules also get added to the generated editorconfig file (currently we only handle formatting and code style options).

> 2. naming:  I would just have GetValueString, or GetOptionValueString.

I thought this would add confusion if the option value itself is a string. Even though the current name is verbose, it clearly describes the semantics of the API.


---
In reply to: [219246165](https://github.com/dotnet/roslyn/pull/30043#discussion_r219246165) [](ancestors = 219246165)

Reply:
> I thought this would add confusion if the option value itself is a string. Even though the current name is verbose, it clearly describes the semantics of the API.

Note: do you actually need the key ever indepedently?  Or do people call into this sole to generate hte `key=value` text?  If the latter, we could just make this function just return that to the caller and leave the caller out of the job of figuring out how to stitch things together.

I personally like that because then that model woudl still work for NamingStyle conventions.  

What do you think?

Reply:
Filed [https://github.com/dotnet/roslyn/issues/30086](https://github.com/dotnet/roslyn/issues/30086) to track adding support for naming rules.

---
In reply to: [219322575](https://github.com/dotnet/roslyn/pull/30043#discussion_r219322575) [](ancestors = 219322575,219322352,219315803)